### ISSUE TITLE
CloudBuildTrigger has trouble updating after v1.33.0

### ISSUE THREAD

**Issue by tonybenchsci** [GoogleCloudPlatform/k8s-config-connector#357]
**Describe the bug**
Any changes to CloudBuildTrigger `spec.github` will return 
```
"Update call failed: error applying desired state: summary: Error updating
      Trigger \"projects/b6i-cicd/triggers/00d81165-513f-4c20-b722-f93e9823acac\":
      googleapi: Error 400: Invalid value at 'trigger.github' (oneof), oneof field
      'event' is already set. Cannot set 'push'\nDetails:\n[\n  {\n    \"@type\":
      \"type.googleapis.com/google.rpc.BadRequest\",\n    \"fieldViolations\": [\n
      \     {\n        \"description\": \"Invalid value at 'trigger.github' (oneof),
      oneof field 'event' is already set. Cannot set 'push'\",\n        \"field\":
      \"trigger.github\"\n      }\n    ]\n  }\n], detail: "
```

Behaviour suggests that KCC merges the mappings (key-value) rather than overwrite, and then complains that there are two 'event' defined.

**ConfigConnector Version**
1.33.0 (previously on 1.26.0)

**To Reproduce**
Create a CloudBuildTrigger, then change the YAML from, say triggering on push to on pull.

*YAML snippets:*

```yaml
apiVersion: cloudbuild.cnrm.cloud.google.com/v1beta1
kind: CloudBuildTrigger
metadata:
  name: frontend-e2e-tests
spec:
  description: End-to-end tests on frontend pull requests.
  disabled: true
  filename: frontend/cloudbuild_e2e_tests.yaml
  github:
    owner: <BLAH>
    name:  <BLAH>
    pullRequest:
      branch: .*
  includedFiles:
  # Defines the frontend.
  - "frontend/**"

```

**Comment by spew** [comment]
Hi @tonybenchsci thanks for reporting this, we will work on reproducing and let you know.

**Comment by spew** [comment]
hi @tonybenchsci can you let me know how this resource was created, did you "acquire" a resource and it had 'push' defined? Or did you specify the full resource with push and deleted it and added pullRequest?

**Comment by tonybenchsci** [comment]
Hi @spew, the full resource was created with `github.push` at first (status UpToDate). We tried to change it to `github.pullRequest`, which led to the error.

**Comment by spew** [comment]
Thanks for the information @tonybenchsci . Was it created with github.push via config connector? That is to say the resource was entirely created and managed through config connector and you managed to reach this state?

**Comment by tonybenchsci** [comment]
@spew Yes it was created initially with KCC as well. We pretty much using it for our entire GCP infra :)

**Comment by spew** [comment]
Thank you, we will be taking a look at this one.

**Comment by jcanseco** [comment]
Hello all, we recently fixed this in KCC 1.78.0. Closing.
